src/Platform/Microsoft.Testing.Platform/OutputDevice/Terminal/TerminalTestReporterCommandLineOptionsProvider.cs

```diff
@@ -32,12 +33,21 @@ internal sealed class TerminalTestReporterCommandLineOptionsProvider : ICommandL
     public IReadOnlyCollection<CommandLineOption> GetCommandLineOptions()
         =>
         [
-            new(NoProgressOption, PlatformResources.TerminalNoProgressOptionDescription, ArgumentArity.Zero, false),
-            new(NoAnsiOption, PlatformResources.TerminalNoAnsiOptionDescription, ArgumentArity.Zero, false)
+            new(NoProgressOption, PlatformResources.TerminalNoProgressOptionDescription, ArgumentArity.Zero, isHidden: false),
+            new(NoAnsiOption, PlatformResources.TerminalNoAnsiOptionDescription, ArgumentArity.Zero, isHidden: false),
+            new(OutputOption, PlatformResources.TerminalOutputOptionDescription, ArgumentArity.ExactlyOne, isHidden: false),
         ];
 
     public Task<ValidationResult> ValidateOptionArgumentsAsync(CommandLineOption commandOption, string[] arguments)
-        => ValidationResult.ValidTask;
+        => commandOption.Name switch
+        {
+            NoProgressOption => ValidationResult.ValidTask,
+            NoAnsiOption => ValidationResult.ValidTask,
+            OutputOption => "normal".Equals(arguments[0], StringComparison.OrdinalIgnoreCase) || "detailed".Equals(arguments[0], StringComparison.OrdinalIgnoreCase)
+                ? ValidationResult.ValidTask
+                : ValidationResult.InvalidTask(PlatformResources.TerminalOutputOptionInvalidArgument),
+            _ => throw new NotSupportedException(),
```

Comment:
I think we should use the unreachable state exception here (ApplicationStateGuard type)

Reply:
done.